### ISSUE TITLE
softwareraid with two disk mentioned in the yaml does not exit cleanly. 

### ISSUE THREAD

**Issue by naveedus** [avocado-framework-tests/avocado-misc-tests#454]
softwareraid with two disk mentioned in the yaml does not exit cleanly. 
We see /dev/md-6 is still part of the software raid.

# avocado run softwareraid.py -m softwareraid.py.data/softwareraid.yaml
 avocado run softwareraid.py -m softwareraid.py.data/softwareraid.yaml 
JOB ID     : eaf339b5e0084b99a2162905bcc25f833690e5a5
JOB LOG    : /root/avocado/job-results/job-2017-04-24T06.05-eaf339b/job.log
 (1/6) softwareraid.py:SoftwareRaid.test_run;raidlinear-scenario-f4df: PASS (0.40 s)
 (2/6) softwareraid.py:SoftwareRaid.test_run;raid0-scenario-b22b: PASS (0.27 s)
 (3/6) softwareraid.py:SoftwareRaid.test_run;raid1-scenario-308f: FAIL (0.11 s)
 (4/6) softwareraid.py:SoftwareRaid.test_run;raid5-scenario-99ad: FAIL (0.10 s)
 (5/6) softwareraid.py:SoftwareRaid.test_run;raid10-scenario-9424: FAIL (0.10 s)
 (6/6) softwareraid.py:SoftwareRaid.test_run;raid6-scenario-93a1: FAIL (0.10 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 4 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
TESTS TIME : 1.09 s
# mdadm --detail /dev/md126 
/dev/md126:
           Version : 1.2
        Raid Level : raid0
     Total Devices : 1
       Persistence : Superblock is persistent

             State : inactive

              Name : linux-0a5l:mdsraid  (local to host linux-0a5l)
              UUID : 769cfcd5:d6fe6f19:60bd6c33:336befb9
            Events : 0

    Number   Major   Minor   RaidDevice

       -     254        6        -        /dev/dm-6
# cat softwareraid.py.data/softwareraid.yaml 
scenario:
    disk: /dev/dm-5 /dev/dm-6 
raidlevel: !mux
    raidlinear:
        raid: linear
    raid0:
        raid: 0
    raid1:
        raid: 1
    raid5:
        raid: 5
    raid10:
        raid: 10
    raid6:
        raid: 6

**Comment by PraveenPenguin** [comment]
@vrbagalkote can you please have a look in to this issue ?

**Comment by vrbagalkote** [comment]
@naveedus Can you please attach job.log file.

**Comment by naveedus** [comment]
[job.log.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/972486/job.log.txt)

**Comment by PraveenPenguin** [comment]
@naveedus are you not facing this any more please suggest ?

**Comment by naveedus** [comment]
I uploaded the job log and closed this by mistake.. 
reopening the issue.

**Comment by narasimhan-v** [comment]
The issue here is:
```
2017-04-24 06:05:19,813 process          L0389 INFO | Running 'echo 'yes' | mdadm --create --verbose --assume-clean             /dev/md/mdsraid --level=1 --raid-devices=1 /dev/dm-5             --spare-devices=1 /dev/dm-6 --force'
2017-04-24 06:05:19,818 process          L0479 DEBUG| [stderr] mdadm: /dev/dm-5 appears to be part of a raid array:
2017-04-24 06:05:19,819 process          L0499 INFO | Command 'echo 'yes' | mdadm --create --verbose --assume-clean             /dev/md/mdsraid --level=1 --raid-devices=1 /dev/dm-5             --spare-devices=1 /dev/dm-6 --force' finished with 1 after 0.00364995002747s
2017-04-24 06:05:19,819 process          L0479 DEBUG| [stderr]        level=raid0 devices=2 ctime=Mon Apr 24 06:05:19 2017
2017-04-24 06:05:19,819 process          L0479 DEBUG| [stderr] mdadm: Note: this array has metadata at the start and
2017-04-24 06:05:19,819 process          L0479 DEBUG| [stderr]     may not be suitable as a boot device.  If you plan to
2017-04-24 06:05:19,820 process          L0479 DEBUG| [stderr]     store '/boot' on this device please ensure that
2017-04-24 06:05:19,820 process          L0479 DEBUG| [stderr]     your boot-loader understands md/v1.x metadata, or use
2017-04-24 06:05:19,820 process          L0479 DEBUG| [stderr]     --metadata=0.90
2017-04-24 06:05:19,820 process          L0479 DEBUG| [stderr] mdadm: super1.x cannot open /dev/dm-6: Device or resource busy
2017-04-24 06:05:19,820 process          L0479 DEBUG| [stderr] mdadm: /dev/dm-6 is not suitable for this array.
2017-04-24 06:05:19,820 process          L0479 DEBUG| [stderr] mdadm: create aborted
```

@naveedus Please verify if the PR (https://github.com/avocado-framework-tests/avocado-misc-tests/pull/487) solves the issue.

**Comment by naveedus** [comment]
I verified the issue and now the created raid array is stopped and exits cleanly. 
we can close this issue. 

meantime I found that minimum 3 disks are required to test RAID1, where 3rd disk is used as spare device used for fail-over. 

Interestingly now with 3 disks the test runs RAID 5 test as well which I feel is wrong. 
How can mdadm create RAID5 with two disks?

======console logs snippet====
2017-05-18 06:22:50,686 process          L0499 INFO | Command 'echo 'yes' | mdadm --create --verbose --assume-clean             /dev/md/mdsraid --level=5 --raid-devices=2 /dev/dm-10 /dev/dm-9             --spare-devices=1 /dev/dm-8 --force' finished with 0 after 0.0735051631927s
2017-05-18 06:22:50,686 process          L0389 INFO | Running 'mdadm --detail /dev/md/mdsraid'
2017-05-18 06:22:50,694 process          L0479 DEBUG| [stdout] /dev/md/mdsraid:
2017-05-18 06:22:50,694 process          L0499 INFO | Command 'mdadm --detail /dev/md/mdsraid' finished with 0 after 0.0042929649353s
2017-05-18 06:22:50,694 process          L0479 DEBUG| [stdout]            Version : 1.2
2017-05-18 06:22:50,694 process          L0479 DEBUG| [stdout]      Creation Time : Thu May 18 06:22:50 2017
2017-05-18 06:22:50,694 process          L0479 DEBUG| [stdout]         Raid Level : raid5
2017-05-18 06:22:50,695 process          L0479 DEBUG| [stdout]         Array Size : 378732544 (361.19 GiB 387.82 GB)
2017-05-18 06:22:50,695 process          L0479 DEBUG| [stdout]      Used Dev Size : 378732544 (361.19 GiB 387.82 GB)
2017-05-18 06:22:50,695 process          L0479 DEBUG| [stdout]       Raid Devices : 2
2017-05-18 06:22:50,695 process          L0479 DEBUG| [stdout]      Total Devices : 3
2017-05-18 06:22:50,695 process          L0479 DEBUG| [stdout]        Persistence : Superblock is persistent
2017-05-18 06:22:50,695 process          L0479 DEBUG| [stdout] 
2017-05-18 06:22:50,695 process          L0479 DEBUG| [stdout]      Intent Bitmap : Internal
2017-05-18 06:22:50,696 process          L0479 DEBUG| [stdout] 
2017-05-18 06:22:50,696 process          L0479 DEBUG| [stdout]        Update Time : Thu May 18 06:22:50 2017
2017-05-18 06:22:50,696 process          L0479 DEBUG| [stdout]              State : clean 
2017-05-18 06:22:50,696 process          L0479 DEBUG| [stdout]     Active Devices : 2
2017-05-18 06:22:50,696 process          L0479 DEBUG| [stdout]    Working Devices : 3
2017-05-18 06:22:50,696 process          L0479 DEBUG| [stdout]     Failed Devices : 0
2017-05-18 06:22:50,696 process          L0479 DEBUG| [stdout]      Spare Devices : 1
2017-05-18 06:22:50,697 process          L0479 DEBUG| [stdout] 
2017-05-18 06:22:50,697 process          L0479 DEBUG| [stdout]             Layout : left-symmetric
2017-05-18 06:22:50,697 process          L0479 DEBUG| [stdout]         Chunk Size : 512K
2017-05-18 06:22:50,697 process          L0479 DEBUG| [stdout] 
2017-05-18 06:22:50,697 process          L0479 DEBUG| [stdout] Consistency Policy : bitmap
2017-05-18 06:22:50,697 process          L0479 DEBUG| [stdout] 
2017-05-18 06:22:50,697 process          L0479 DEBUG| [stdout]               Name : linux-0a5l:mdsraid  (local to host linux-0a5l)
2017-05-18 06:22:50,698 process          L0479 DEBUG| [stdout]               UUID : e0214914:eeb25c1d:8d7644de:af154f3a
2017-05-18 06:22:50,698 process          L0479 DEBUG| [stdout]             Events : 0
2017-05-18 06:22:50,698 process          L0479 DEBUG| [stdout] 
2017-05-18 06:22:50,698 process          L0479 DEBUG| [stdout]     Number   Major   Minor   RaidDevice State
2017-05-18 06:22:50,698 process          L0479 DEBUG| [stdout]        0     254       10        0      active sync   /dev/dm-10
2017-05-18 06:22:50,698 process          L0479 DEBUG| [stdout]        1     254        9        1      active sync   /dev/dm-9
2017-05-18 06:22:50,698 process          L0479 DEBUG| [stdout] 
2017-05-18 06:22:50,699 process          L0479 DEBUG| [stdout]        2     254        8        -      spare   /dev/dm-8
2017-05-18 06:22:50,699 process          L0389 INFO | Running 'mdadm --fail /dev/md/mdsraid /dev/dm-9'
2017-05-18 06:22:50,702 process          L0479 DEBUG| [stderr] mdadm: set /dev/dm-9 faulty in /dev/md/mdsraid
2017-05-18 06:22:50,702 process          L0499 INFO | Command 'mdadm --fail /dev/md/mdsraid /dev/dm-9' finished with 0 after 0.00182104110718s
2017-05-18 06:22:50,703 process          L0389 INFO | Running 'mdadm --detail /dev/md/mdsraid'
2017-05-18 06:22:50,708 process          L0479 DEBUG| [stdout] /dev/md/mdsraid:
2017-05-18 06:22:50,708 process          L0499 INFO | Command 'mdadm --detail /dev/md/mdsraid' finished with 0 after 0.00380706787109s
2017-05-18 06:22:50,709 process          L0479 DEBUG| [stdout]            Version : 1.2
2017-05-18 06:22:50,709 process          L0479 DEBUG| [stdout]      Creation Time : Thu May 18 06:22:50 2017
2017-05-18 06:22:50,709 process          L0479 DEBUG| [stdout]         Raid Level : raid5
2017-05-18 06:22:50,709 process          L0479 DEBUG| [stdout]         Array Size : 378732544 (361.19 GiB 387.82 GB)
2017-05-18 06:22:50,709 process          L0479 DEBUG| [stdout]      Used Dev Size : 378732544 (361.19 GiB 387.82 GB)
2017-05-18 06:22:50,709 process          L0479 DEBUG| [stdout]       Raid Devices : 2
2017-05-18 06:22:50,710 process          L0479 DEBUG| [stdout]      Total Devices : 3
2017-05-18 06:22:50,710 process          L0479 DEBUG| [stdout]        Persistence : Superblock is persistent
2017-05-18 06:22:50,710 process          L0479 DEBUG| [stdout] 
2017-05-18 06:22:50,710 process          L0479 DEBUG| [stdout]      Intent Bitmap : Internal
2017-05-18 06:22:50,710 process          L0479 DEBUG| [stdout] 
2017-05-18 06:22:50,710 process          L0479 DEBUG| [stdout]        Update Time : Thu May 18 06:22:50 2017
2017-05-18 06:22:50,711 process          L0479 DEBUG| [stdout]              State : clean, degraded 
2017-05-18 06:22:50,711 process          L0479 DEBUG| [stdout]     Active Devices : 1
2017-05-18 06:22:50,711 process          L0479 DEBUG| [stdout]    Working Devices : 2
2017-05-18 06:22:50,711 process          L0479 DEBUG| [stdout]     Failed Devices : 1
2017-05-18 06:22:50,711 process          L0479 DEBUG| [stdout]      Spare Devices : 1
2017-05-18 06:22:50,711 process          L0479 DEBUG| [stdout] 
2017-05-18 06:22:50,711 process          L0479 DEBUG| [stdout]             Layout : left-symmetric
2017-05-18 06:22:50,712 process          L0479 DEBUG| [stdout]         Chunk Size : 512K
2017-05-18 06:22:50,712 process          L0479 DEBUG| [stdout] 
2017-05-18 06:22:50,712 process          L0479 DEBUG| [stdout] Consistency Policy : bitmap
2017-05-18 06:22:50,712 process          L0479 DEBUG| [stdout] 
2017-05-18 06:22:50,712 process          L0479 DEBUG| [stdout]               Name : linux-0a5l:mdsraid  (local to host linux-0a5l)
2017-05-18 06:22:50,712 process          L0479 DEBUG| [stdout]               UUID : e0214914:eeb25c1d:8d7644de:af154f3a
2017-05-18 06:22:50,712 process          L0479 DEBUG| [stdout]             Events : 1
2017-05-18 06:22:50,713 process          L0479 DEBUG| [stdout] 
2017-05-18 06:22:50,713 process          L0479 DEBUG| [stdout]     Number   Major   Minor   RaidDevice State
2017-05-18 06:22:50,713 process          L0479 DEBUG| [stdout]        0     254       10        0      active sync   /dev/dm-10
2017-05-18 06:22:50,713 process          L0479 DEBUG| [stdout]        1     254        9        1      faulty   /dev/dm-9
2017-05-18 06:22:50,713 process          L0479 DEBUG| [stdout] 
2017-05-18 06:22:50,713 process          L0479 DEBUG| [stdout]        2     254        8        -      spare   /dev/dm-8
2017-05-18 06:22:50,714 process          L0389 INFO | Running 'mdadm --manage /dev/md/mdsraid --remove /dev/dm-9'
2017-05-18 06:22:50,810 process          L0479 DEBUG| [stderr] mdadm: hot removed /dev/dm-9 from /dev/md/mdsraid
2017-05-18 06:22:50,810 process          L0499 INFO | Command 'mdadm --manage /dev/md/mdsraid --remove /dev/dm-9' finished with 0 after 0.0948021411896s
2017-05-18 06:22:50,811 process          L0389 INFO | Running 'mdadm --detail /dev/md/mdsraid'
2017-05-18 06:22:50,829 process          L0479 DEBUG| [stdout] /dev/md/mdsraid:
2017-05-18 06:22:50,829 process          L0499 INFO | Command 'mdadm --detail /dev/md/mdsraid' finished with 0 after 0.016893863678s
2017-05-18 06:22:50,830 process          L0479 DEBUG| [stdout]            Version : 1.2
2017-05-18 06:22:50,830 process          L0479 DEBUG| [stdout]      Creation Time : Thu May 18 06:22:50 2017
2017-05-18 06:22:50,831 process          L0479 DEBUG| [stdout]         Raid Level : raid5
2017-05-18 06:22:50,831 process          L0479 DEBUG| [stdout]         Array Size : 378732544 (361.19 GiB 387.82 GB)
2017-05-18 06:22:50,831 process          L0479 DEBUG| [stdout]      Used Dev Size : 378732544 (361.19 GiB 387.82 GB)
2017-05-18 06:22:50,831 process          L0479 DEBUG| [stdout]       Raid Devices : 2
2017-05-18 06:22:50,831 process          L0479 DEBUG| [stdout]      Total Devices : 2
2017-05-18 06:22:50,832 process          L0479 DEBUG| [stdout]        Persistence : Superblock is persistent
2017-05-18 06:22:50,832 process          L0479 DEBUG| [stdout] 
2017-05-18 06:22:50,832 process          L0479 DEBUG| [stdout]      Intent Bitmap : Internal
2017-05-18 06:22:50,832 process          L0479 DEBUG| [stdout] 
2017-05-18 06:22:50,832 process          L0479 DEBUG| [stdout]        Update Time : Thu May 18 06:22:50 2017
2017-05-18 06:22:50,832 process          L0479 DEBUG| [stdout]              State : clean, degraded, recovering 
2017-05-18 06:22:50,833 process          L0479 DEBUG| [stdout]     Active Devices : 1
2017-05-18 06:22:50,833 process          L0479 DEBUG| [stdout]    Working Devices : 2
2017-05-18 06:22:50,833 process          L0479 DEBUG| [stdout]     Failed Devices : 0
2017-05-18 06:22:50,833 process          L0479 DEBUG| [stdout]      Spare Devices : 1
2017-05-18 06:22:50,833 process          L0479 DEBUG| [stdout] 
2017-05-18 06:22:50,833 process          L0479 DEBUG| [stdout]             Layout : left-symmetric
2017-05-18 06:22:50,834 process          L0479 DEBUG| [stdout]         Chunk Size : 512K
2017-05-18 06:22:50,834 process          L0479 DEBUG| [stdout] 
2017-05-18 06:22:50,834 process          L0479 DEBUG| [stdout] Consistency Policy : bitmap
2017-05-18 06:22:50,834 process          L0479 DEBUG| [stdout] 
2017-05-18 06:22:50,834 process          L0479 DEBUG| [stdout]     Rebuild Status : 0% complete
2017-05-18 06:22:50,834 process          L0479 DEBUG| [stdout] 
2017-05-18 06:22:50,835 process          L0479 DEBUG| [stdout]               Name : linux-0a5l:mdsraid  (local to host linux-0a5l)
2017-05-18 06:22:50,835 process          L0479 DEBUG| [stdout]               UUID : e0214914:eeb25c1d:8d7644de:af154f3a
2017-05-18 06:22:50,835 process          L0479 DEBUG| [stdout]             Events : 2
2017-05-18 06:22:50,835 process          L0479 DEBUG| [stdout] 
2017-05-18 06:22:50,835 process          L0479 DEBUG| [stdout]     Number   Major   Minor   RaidDevice State
2017-05-18 06:22:50,835 process          L0479 DEBUG| [stdout]        0     254       10        0      active sync   /dev/dm-10
2017-05-18 06:22:50,836 process          L0479 DEBUG| [stdout]        2     254        8        1      spare rebuilding   /dev/dm-8
2017-05-18 06:22:50,836 process          L0389 INFO | Running 'mdadm --manage /dev/md/mdsraid --add /dev/dm-9'
2017-05-18 06:22:50,876 process          L0499 INFO | Command 'mdadm --manage /dev/md/mdsraid --add /dev/dm-9' finished with 0 after 0.037789106369s
2017-05-18 06:22:50,876 process          L0479 DEBUG| [stderr] mdadm: re-added /dev/dm-9
2017-05-18 06:22:50,877 process          L0389 INFO | Running 'mdadm --detail /dev/md/mdsraid'
2017-05-18 06:22:50,994 process          L0479 DEBUG| [stdout] /dev/md/mdsraid:
2017-05-18 06:22:50,994 process          L0499 INFO | Command 'mdadm --detail /dev/md/mdsraid' finished with 0 after 0.115511894226s
2017-05-18 06:22:50,995 process          L0479 DEBUG| [stdout]            Version : 1.2
2017-05-18 06:22:50,995 process          L0479 DEBUG| [stdout]      Creation Time : Thu May 18 06:22:50 2017
2017-05-18 06:22:50,995 process          L0479 DEBUG| [stdout]         Raid Level : raid5
2017-05-18 06:22:50,995 process          L0479 DEBUG| [stdout]         Array Size : 378732544 (361.19 GiB 387.82 GB)
2017-05-18 06:22:50,996 process          L0479 DEBUG| [stdout]      Used Dev Size : 378732544 (361.19 GiB 387.82 GB)
2017-05-18 06:22:50,996 process          L0479 DEBUG| [stdout]       Raid Devices : 2
2017-05-18 06:22:50,996 process          L0479 DEBUG| [stdout]      Total Devices : 3
2017-05-18 06:22:50,996 process          L0479 DEBUG| [stdout]        Persistence : Superblock is persistent
2017-05-18 06:22:50,996 process          L0479 DEBUG| [stdout] 
2017-05-18 06:22:50,996 process          L0479 DEBUG| [stdout]      Intent Bitmap : Internal
2017-05-18 06:22:50,997 process          L0479 DEBUG| [stdout] 
2017-05-18 06:22:50,997 process          L0479 DEBUG| [stdout]        Update Time : Thu May 18 06:22:50 2017
2017-05-18 06:22:50,997 process          L0479 DEBUG| [stdout]              State : clean, degraded, recovering 
2017-05-18 06:22:50,997 process          L0479 DEBUG| [stdout]     Active Devices : 1
2017-05-18 06:22:50,997 process          L0479 DEBUG| [stdout]    Working Devices : 3
2017-05-18 06:22:50,998 process          L0479 DEBUG| [stdout]     Failed Devices : 0
2017-05-18 06:22:50,998 process          L0479 DEBUG| [stdout]      Spare Devices : 2
2017-05-18 06:22:50,998 process          L0479 DEBUG| [stdout] 
2017-05-18 06:22:50,998 process          L0479 DEBUG| [stdout]             Layout : left-symmetric
2017-05-18 06:22:50,998 process          L0479 DEBUG| [stdout]         Chunk Size : 512K
2017-05-18 06:22:50,998 process          L0479 DEBUG| [stdout] 
2017-05-18 06:22:50,999 process          L0479 DEBUG| [stdout] Consistency Policy : bitmap
2017-05-18 06:22:50,999 process          L0479 DEBUG| [stdout] 
2017-05-18 06:22:50,999 process          L0479 DEBUG| [stdout]     Rebuild Status : 0% complete
2017-05-18 06:22:50,999 process          L0479 DEBUG| [stdout] 
2017-05-18 06:22:50,999 process          L0479 DEBUG| [stdout]               Name : linux-0a5l:mdsraid  (local to host linux-0a5l)
2017-05-18 06:22:50,999 process          L0479 DEBUG| [stdout]               UUID : e0214914:eeb25c1d:8d7644de:af154f3a
2017-05-18 06:22:51,000 process          L0479 DEBUG| [stdout]             Events : 5
2017-05-18 06:22:51,000 process          L0479 DEBUG| [stdout] 
2017-05-18 06:22:51,000 process          L0479 DEBUG| [stdout]     Number   Major   Minor   RaidDevice State
2017-05-18 06:22:51,001 process          L0479 DEBUG| [stdout]        0     254       10        0      active sync   /dev/dm-10
2017-05-18 06:22:51,001 process          L0479 DEBUG| [stdout]        2     254        8        1      spare rebuilding   /dev/dm-8
2017-05-18 06:22:51,001 process          L0479 DEBUG| [stdout] 
2017-05-18 06:22:51,001 process          L0479 DEBUG| [stdout]        1     254        9        -      spare   /dev/dm-9
2017-05-18 06:22:51,002 process          L0389 INFO | Running 'mdadm --manage /dev/md/mdsraid --stop'
2017-05-18 06:22:51,230 process          L0479 DEBUG| [stderr] mdadm: stopped /dev/md/mdsraid
2017-05-18 06:22:51,230 process          L0499 INFO | Command 'mdadm --manage /dev/md/mdsraid --stop' finished with 0 after 0.22633600235s
2017-05-18 06:22:51,231 process          L0389 INFO | Running 'mdadm --assemble /dev/md/mdsraid /dev/dm-10 /dev/dm-9 /dev/dm-8'
2017-05-18 06:22:51,320 process          L0479 DEBUG| [stderr] mdadm: /dev/md/mdsraid has been started with 1 drive (out of 2), 1 rebuilding and 1 spare.
2017-05-18 06:22:51,384 process          L0499 INFO | Command 'mdadm --assemble /dev/md/mdsraid /dev/dm-10 /dev/dm-9 /dev/dm-8' finished with 0 after 0.151476860046s
2017-05-18 06:22:51,384 process          L0389 INFO | Running 'mdadm --detail /dev/md/mdsraid | grep State | grep recovering'
2017-05-18 06:22:51,475 process          L0479 DEBUG| [stdout]              State : clean, degraded, recovering
